pip/utils/glibc.py

```diff
@@ -75,7 +74,10 @@ def have_compatible_glibc(required_major, minimum_minor):
 def libc_ver():
     glibc_version = glibc_version_string()
     if glibc_version is None:
-        # For non-glibc platforms, fall back on platform.libc_ver
-        return platform.libc_ver()
+        # At least as of Python 3.6a2, platform.libc_ver() doesn't do anything
+        # meaningful with non-glibc versions anyway, so there's no point in
+        # even falling back on it -- pretty much all it would do is give us
+        # the opportunity to get more corrupt data.
+        ("not-glibc", "")
```

Comment:
I guess it depends on whether we're optimizing for database aesthetics or database usability? :-)

If it helps, it looks like musl doesn't provide any way to get this information at all :-(